src/core/triggers.systemd.in

```diff
@@ -23,136 +23,71 @@
-        posix.wait(pid)
-    end
-end
+%transfiletriggerun -- @systemunitdir@ /etc/systemd/system
+# On removal, we need to run daemon-reload after any units have been
+# removed. %transfiletriggerpostun would be ideal, but it does not get
+# executed for some reason.
+# On upgrade, we need to run daemon-reload after any new unit files
+# have been installed, but before %postun scripts in packages get
+# executed. %transfiletriggerun gets the right list of files
+# but it is invoked too early (before changes happen).
+# %filetriggerpostun happens at the right time, but it fires for
+# every package.
+# To execute the reload at the right time, we create a state
+# file in %transfiletriggerun and execute the daemon-reload in
+# the first %filetriggerpostun.
 
-%transfiletriggerun -p <lua> -- @systemunitdir@ /etc/systemd/system
--- On removal, we need to run daemon-reload after any units have been
--- removed. %transfiletriggerpostun would be ideal, but it does not get
--- executed for some reason.
--- On upgrade, we need to run daemon-reload after any new unit files
--- have been installed, but before %postun scripts in packages get
--- executed. %transfiletriggerun gets the right list of files
--- but it is invoked too early (before changes happen).
--- %filetriggerpostun happens at the right time, but it fires for
--- every package.
--- To execute the reload at the right time, we create a state
--- file in %transfiletriggerun and execute the daemon-reload in
--- the first %filetriggerpostun.
+if test -d "/run/systemd/system"; then
+    mkdir -p %{_localstatedir}/lib/rpm-state/systemd"
+    touch %{_localstatedir}/lib/rpm-state/systemd/needs-reload"
+fi
 
-if posix.access("/run/systemd/system") then
-    posix.mkdir("%{_localstatedir}/lib")
-    posix.mkdir("%{_localstatedir}/lib/rpm-state")
-    posix.mkdir("%{_localstatedir}/lib/rpm-state/systemd")
-    io.open("%{_localstatedir}/lib/rpm-state/systemd/needs-reload", "w")
-end
+%filetriggerpostun -P 1000100 -- @systemunitdir@ /etc/systemd/system
+if test -f "%{_localstatedir}/lib/rpm-state/systemd/needs-reload"; then
+    rm -rf "%{_localstatedir}/lib/rpm-state/systemd"
+    %{_bindir}/systemctl daemon-reload
+fi
 
-%filetriggerpostun -P 1000100 -p <lua> -- @systemunitdir@ /etc/systemd/system
-if posix.access("%{_localstatedir}/lib/rpm-state/systemd/needs-reload") then
-    posix.unlink("%{_localstatedir}/lib/rpm-state/systemd/needs-reload")
-    posix.rmdir("%{_localstatedir}/lib/rpm-state/systemd")
-    pid = posix.fork()
-    if pid == 0 then
-        assert(posix.exec("%{_bindir}/systemctl", "daemon-reload"))
-    elseif pid > 0 then
-        posix.wait(pid)
-    end
-end
+%transfiletriggerin -P 100700 -- @sysusersdir@
+# This script will process files installed in @sysusersdir@ to create
+# specified users automatically. The priority is set such that it
+# will run before the tmpfiles file trigger.
+if test -d /run/systemd/system; then %{_bindir}/systemd-sysusers; fi
```

Comment:
as above, why compress this into one line?

Reply:
i mean, if you want to compress this into one line, why not do:

test -d … && …?

Reply:
> test -d … && …?

I actually did it that way to start, but then remember in shell that will result in `exit 1` and hence RPM complaining about script failures...  (and on a related note, in rpm-ostree by default script failures are fatal)

Reply:
ok